### PR TITLE
Remove `SeededTest` class

### LINKED ISSUE #6234: Remove `SeededTest` class
This class is not that relevant now that our sampling routines do not rely on global seeding. It causes some confusions because it seems like the tests within should be seeded by default (they used to)

The only few uses that I think are still relevant is when we use it to obtain a seed or create fake data with legacy numpy random routines (i.e., not Generators). I think we can just refactor it to not rely on a base Test class.

### PR DESCRIPTION
**Pull request by aerubanov**
This PR remove `SeededTest` class and close #6234 

----
:books: Documentation preview :books:: https://pymc--6799.org.readthedocs.build/en/6799/

### CODE REVIEW

**Comment by aerubanov** [comment]
Some tests fails after removing `SeededTest` and I`m working on fixing it.

**Comment by aerubanov** [comment]
@ricardoV94 , @twiecki Could you please review this PR when you will have some time?

File: tests/conftest.py [hunk 1]
```diff
@@ -44,4 +44,4 @@ def strict_float32():
 @pytest.fixture(scope="function", autouse=False)
 def seeded_test():
     # TODO: use this instead of SeededTest
-    np.random.seed(42)
+    np.random.seed(20160911)
```
Comment (ricardoV94) [inline tests/conftest.py:47]: Any reason for this change?

File: tests/conftest.py [hunk 2]
```diff
@@ -44,4 +44,4 @@ def strict_float32():
 @pytest.fixture(scope="function", autouse=False)
 def seeded_test():
     # TODO: use this instead of SeededTest
```
Comment (ricardoV94) [inline tests/conftest.py]:
```suggestion
```

**Review by ricardoV94 — COMMENTED** [review]
Looks pretty good. Small question and suggestion below

File: tests/conftest.py (hunk 1 above)
Comment (aerubanov) [reply tests/conftest.py:47]: I found some test that fails with random seed 42 - the result differs from expected just a little bit more than the selected threshold value. That is why I decided to use here value from the `SeededTest` class for random seed.

**Review by ricardoV94 — APPROVED** [review]

**Comment by ferrine** [comment]
Why not using a fixture with autouse?

**Comment by ricardoV94** [comment]
> Why not using a fixture with autouse?

For what? The point in that global seeding doesn't do anything in PyMC anymore so each test should create a generator/seed to pass to the functions that need seeding.

Do you mean a fixture to create such a seed?

**Comment by aerubanov** [comment]
@ricardoV94 I think @ferrine ask about pytest fixture [autouse](https://docs.pytest.org/en/6.2.x/fixture.html#autouse-fixtures-fixtures-you-don-t-have-to-request) argument to apply `seeded_test` fixture to all tests.

**Comment by ricardoV94** [comment]
> @ricardoV94 I think @ferrine ask about pytest fixture [autouse](https://docs.pytest.org/en/6.2.x/fixture.html#autouse-fixtures-fixtures-you-don-t-have-to-request) argument to apply `seeded_test` fixture to all tests.

I don't think we should do that. If anything we should move away from using any numpy RandomState stuff, so the fixture would be useless.

**Comment by ricardoV94** [comment]
There's a conflict with main now. Otherwise I think it's good to merge

**Comment by ricardoV94** [comment]
Thanks @aerubanov!

**Pull request merged**